### ISSUE TITLE
Support env, volumes, CPU limits and endpoints for every Devfile component type

### ISSUE THREAD

**Issue by l0rd** [eclipse-che/che#14970]
### Is your enhancement related to a problem?

Currently only components of type `dockerImage` support `env`, `volumes` and `endpoint` attributes. This is limiting the possible workspace scenarios using components of types kubernetes, openshift and che plugins.

### Describe the solution you'd like

- [x] Support `env` for components of type kubernetes/openshift #15121
- [x] Support `env` for components of type chePlugin and cheEditor #14343
- [x] Provide an ability to override CPU limits of plugin/editor containers in devfile | https://github.com/eclipse/che/issues/15306
- [x] Support `volumes` for components of type kubernetes/openshift #15122
- [x] Support `volumes` for components of type chePlugin and cheEditor #13318 (also discussed in #14634) 
- [x] Support `endpoints` for components of type kubernetes/openshift #15124
- [x] Support `endpoints` for components of type chePlugin and cheEditor #15126
- [x] Provide an ability to override CPU limits of dockerimage containers in devfile #16588

**Comment by mmorhun** [comment]
Is there any plans to add support of described above features for Che editor? (Just faced a need to set an environment variable for Che Theia, but it is not possible without changing its meta.yaml).

**Comment by l0rd** [comment]
@mmorhun good point

**Comment by sunix** [comment]
Hello, looks like there is a regression on that one. I used to have a workspace running with cpulimit set to 6 for the entire pod. But it seems like in the last versions, cpu request are set to 1 for the plugins and 2 for any other containers. my workspace needs max cpu set to 16 for the entire pod now.

I have tried CHE_WORKSPACE_DEFAULT__CPU__LIMIT__CORES, but it seems to drastically slow down workspace startup.

I have a lot of component of type 'dockerimage' in my devfile. And this is not allowing me to optimize cpu usage per container as I would need to. I am wondering how we could sort this out to avoid having the user to go into the cpu optimization hell.
